// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
syntax = "proto3";

package chrome_browser_media.proto;

// Chrome requires this.
option optimize_for = LITE_RUNTIME;

// This list of capabilities should be kept in sync with the bit-mask found in
// CastMediaSink.capabilities
// chromium/src/components/media_router/common/mojom/media_router.mojom
message DeviceCapabilities {
  bool video_out = 1;
  bool video_in = 2;
  bool audio_out = 3;
  bool audio_in = 4;
  bool dev_mode = 5;
}

message NetworkInfo {
  string host_name = 1;
  string port = 2;
  string ip_v4_address = 3;
  string ip_v6_address = 4;
}

// The device's casting data that is returned in a DiscoverReceiverRequest
// google3/google/internal/chrome/castedu/messaging/v1/discovery_service.proto
message DiscoveryDevice {
  string display_name = 1;
  DeviceCapabilities device_capabilities = 4;
  NetworkInfo network_info = 5;
  string id = 6;
}
